File: src/1773.count-items-matching-a-rule.1899/solution.ts
function countMatches (items: string[][], ruleKey: string, ruleValue: string): number {
    let result = 0;
    for (const [type, color, name, ] of items) {
        switch (ruleKey) {
        case 'type':
            if (type === ruleValue) {
                result++;
            }
            break;
        case 'color':
            if (color === ruleValue) {
                result++;
            }
            break;
        case 'name':
            if (name === ruleValue) {
                result++;
            }
            break;
        }
    }
    return result;
}

File: src/1773.count-items-matching-a-rule.1899/solution.java
class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int result = 0;
        int index = 0;
        if (ruleKey.equals("color")) {
            index = 1;
        } else if (ruleKey.equals("name")) {
            index = 2;
        }

        for (List<String> item : items) {
            if (item.get(index).equals(ruleValue)) {
                result++;
            }
        }
        return result;
    }
}